src/sentry/api/event_search.py

```diff
@@ -1266,16 +1268,31 @@ def get_filter(query=None, params=None):
 }
 
 
-def get_json_meta_type(field_alias, snuba_type):
+def get_json_meta_type(field_alias, snuba_type, field=None):
     alias_definition = FIELD_ALIASES.get(field_alias)
     if alias_definition and alias_definition.get("result_type"):
         return alias_definition.get("result_type")
+
     snuba_json = get_json_type(snuba_type)
-    function_match = FUNCTION_ALIAS_PATTERN.match(field_alias)
-    if function_match and snuba_json != "string":
-        function_definition = FUNCTIONS.get(function_match.group(1))
-        if function_definition and function_definition.result_type:
-            return function_definition.result_type
+    if snuba_json != "string":
+        if field is not None:
+            function_match = FUNCTION_PATTERN.search(field)
+            if function_match:
+                function_name, columns = parse_function(field, function_match)
+                function_definition = FUNCTIONS.get(function_name)
+                if function_definition:
+                    result_type = function_definition.get_result_type(field, columns)
+                    if result_type is not None:
+                        return result_type
+
+        function_match = FUNCTION_ALIAS_PATTERN.match(field_alias)
+        if function_match:
+            function_definition = FUNCTIONS.get(function_match.group(1))
+            if function_definition:
+                result_type = function_definition.get_result_type()
```

Comment:
This block appears twice should it be extracted into a helper function? Something like `get_function_def(fieldname)`? That function could return None or the Function definition object.

Reply:
this has been refactored and no longer applies